googleapiclient/http.py

```diff
@@ -799,16 +815,24 @@ def next_chunk(self, http=None, num_retries=0):
         start_headers['X-Upload-Content-Length'] = size
       start_headers['content-length'] = str(self.body_size)
 
+      resp = None
       for retry_num in range(num_retries + 1):
         if retry_num > 0:
           self._sleep(self._rand() * 2**retry_num)
           logging.warning(
-              'Retry #%d for resumable URI request: %s %s, following status: %d'
-              % (retry_num, self.method, self.uri, resp.status))
-
-        resp, content = http.request(self.uri, method=self.method,
-                                     body=self.body,
-                                     headers=start_headers)
+              'Retry #%d for resumable URI request: %s %s%s'
+              % (retry_num, self.method, self.uri,
+                 ', following status: %d' % resp.status if resp else ''))
+
+        try:
+          resp, content = http.request(self.uri, method=self.method,
+                                       body=self.body,
+                                       headers=start_headers)
+        except ssl.SSLError:
```

Comment:
This logic appears nearly identically in three places - please factor it out into a module-scope module-private helper function?